### ISSUE TITLE
install vs build - libupnp path broken

### ISSUE THREAD

**Issue by whorfin** [v00d00/gerbera#57]
I didn't notice this at first as i've been manually installing the binary, but finally traced this down:

in the build dir,
ldd gerbera
[all the lovely paths are correct]

ldd  CMakeFiles/CMakeRelink.dir/gerbera
libupnp and libixml show as (file not found)

The CMakeFiles/gerbera.dir/relink.txt file is missing the "-Wl,-R..." paths found in link.txt, and thus the
loader paths are missing

It's not obvious to me how the cmake config is causing this to happen, but it is, and it's a problem.

**Comment by v00d00** [comment]
Hmm, I dont appear to even have a `CMakeRelink.dir`

**Comment by v00d00** [comment]
How did you get the relink.txt? I dont have one it seems.

**Comment by whorfin** [comment]
$ make install

I'd been making and installing binaries manually 
After the big rename I did a "make install" again and saw this problem 

Plain make uses link.txt
make install doesn't just install the binary from the build dir, it makes a unique binary from a different relink.txt file and installs that.  Joy

**Comment by v00d00** [comment]
Can you paste your full build log? Maybe we need set RPATH on Solaris?

**Comment by whorfin** [comment]
uh, there's isn't really a build log w/ this cmake setup

Taking a step back, i totally nuked my build dir and rebuilt from scratch.
This time, no CMakeFiles/CMakeRelink.dir was created, and no relink.txt
Same ultimate behavior - installed gerbera missing rpath, can't find libpupnp

This time a clue in "make install" logging:
Install the project...
-- Install configuration: ""
-- Installing: /usr/local/bin/gerbera
-- Set runtime path of "/usr/local/bin/gerbera" to ""

[Sounds like difference between cmake knowing it can use its internal ELF editor vs not; in either case, runtime path for install is empty]

I can see in cmake_install.cmake a call to file(RPATH_CHANGE which is hosing me:
 file(RPATH_CHANGE                                                                                                                                                                                                                                                  FILE "$ENV{DESTDIR}${CMAKE_INSTALL_PREFIX}/bin/gerbera"
         OLD_RPATH "/opt/local/lib:/usr/lib/64:/usr/local/lib:"
         NEW_RPATH "")

Then i found this:
https://cmake.org/Wiki/CMake_RPATH_handling

So TL;DR - if I add "-DCMAKE_INSTALL_RPATH_USE_LINK_PATH=TRUE" to the cmake command used to generate my build environment, everything works swell.

I'd be tempted to add this
# add the automatically determined parts of the RPATH
# which point to directories outside the build tree to the install RPATH
SET(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)

to CMakeLists.txt, but have no idea if that will hose others, so I won't yet

FYI and FWIW

**Comment by v00d00** [comment]
Can you try building with CMAKE_BUILD_WITH_INSTALL_RPATH=TRUE and see what we get?

**Comment by whorfin** [comment]
If I do that, I get a gerbera in the build dir which is missing the appropriate paths, ie can't find libpupnp, as expected from the documentation above.
CMAKE_INSTALL_RPATH_USE_LINK_PATH works fine

**Comment by v00d00** [comment]
@whorfin thanks for you work on this one! 👍
